Frameworks/UIKit/UITextField.mm

```diff
@@ -322,43 +387,94 @@ - (NSDictionary*)typingAttributes {
 // Properties: Sizing the Text Field�s Text
 //
 /**
- @Status Stub
+ @Status Interoperable
 */
 - (void)setAdjustsFontSizeToFitWidth:(BOOL)adjust {
-    UNIMPLEMENTED();
+    _adjustsFontSizeToFitWidth = adjust;
+    [self _adjustFontSizeToFitWidthOrApply];
 }
 
 /**
- @Status Stub
+ @Status Interoperable
 */
-- (void)adjustsFontSizeToFitWidth:(BOOL)adjust {
-    UNIMPLEMENTED();
+- (BOOL)adjustsFontSizeToFitWidth {
+    return _adjustsFontSizeToFitWidth;
+}
+
+- (void)_adjustFontSizeToFitWidthOrApply {
+    if (self.adjustsFontSizeToFitWidth && (self.minimumFontSize > 0) && (self.minimumFontSize != self.font.pointSize) && self.text && self.text.length) {
+        _adjustedFont = self.font;
+        NSString* passwordString = nil;
+        CGFloat elementWidth = 0.0f;
+        
+        // Measure using the password masking character, not the verbatim text
+        if (self->_secureTextMode) {
+            if (_passwordContentElement == nil) {
+                // Not an error, we just might not be loaded yet.
+                [self _applyFont:self.font];
+                return;
+            }
+            passwordString = [@"" stringByPaddingToLength:self.text.length withString:_passwordBox.passwordChar startingAtIndex:0];
+            elementWidth = _passwordContentElement.actualWidth - _passwordContentElement.padding.left - _passwordContentElement.padding.right;
+        } else {
+            if (_textContentElement == nil) {
+                // Not an error, we just might not be loaded yet.
+                [self _applyFont:self.font];
+                return;
+            }
+            elementWidth = _textContentElement.actualWidth - _textContentElement.padding.left - _textContentElement.padding.right;
+        }
+
+        // Try smaller and smaller fonts until it fits within the size, or mins out
+        while (_adjustedFont.pointSize > self.minimumFontSize) {
+            // Grab the width directly from the ContentElement, to allow the cancel button to be considered.
+            if (self->_secureTextMode) {
+                CGSize size = [passwordString sizeWithFont:_adjustedFont];
+                if ((_passwordContentElement == nil) || (size.width <= elementWidth)) {
+                    break;
+                }
+            } else {
+                CGSize size = [self.text sizeWithFont:_adjustedFont];
```

Comment:
if [self text] returns nil here we'll run into the same issue on arm release until #1365 is resolved.